File: LeetCode/Python/mergeIntervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        length = len(intervals)

        if length <= 1:
            return intervals

        intervals = sorted(intervals, key=lambda x: x[0])
        ret = []
        ret.append(intervals[0])
        for i in range(1, length):
            curr = intervals[i]
            last_ret = ret[len(ret) - 1]

            if last_ret[1] >= curr[0]:
                last_ret[1] = max(curr[1], last_ret[1])
            else:
                ret.append(curr)
        
        return ret

File: LeetCode/C#/MergeIntervals.cs
public class Solution {
    public int[][] Merge(int[][] intervals) {
        
        if(intervals.Length <= 1) {
            return intervals;
        }

        Array.Sort(intervals, (a,b) => a[0] -b[0]);

        var ret = new List<int[]>();
        ret.Add(intervals[0]);

        for(int i = 1; i < intervals.Length; i++) {
            var curr = intervals[i];
            var last = ret[ret.Count - 1];

            if(last[1] >= curr[0]) {
                last[1] = Math.Max(curr[1], last[1]);
            } else {
                ret.Add(curr);
            }
        }

        return ret.ToArray();
    }
}